Add support for wildcard_query function (#156)
Bit-Quill/opensearch-project-sql: docs/user/dql/functions.rst

    |----------------+---------------+-----------------+------------------|
    | DATE           | INTEGER       | DATETIME        | STRUCT           |
    +----------------+---------------+-----------------+------------------+

WILDCARD_QUERY
------------

Description
>>>>>>>>>>>

``wildcard_query(field_expression, query_expression[, option=<option_value>]*)``

> The wildcard_query function maps to the wildcard_query query used in search engine, to return the documents that match a provided text with a given field.

Reviewer: ```suggestion
The wildcard_query function maps to the wildcard_query query used in search engine. It returns documents that match provided text in the specified field.
```

Author: Fixed in 0b9752ca994a334de8e9bbc6bd7ac49fadd37944

Supported wildcard characters can be found here: https://opensearch.org/docs/latest/opensearch/query-dsl/term/#wildcards

Available parameters include:

- boost
- case_insensitive
- rewrite

For backward compatibility, wildcardquery is also supported and mapped to wildcard_query query as well.

Example with only ``field`` and ``query`` expressions, and all other parameters are set default values::

>     os> select Id, Tags from beer where wildcard_query(Tags, 'champagn*');

Author: Fixed in a2ca906329ab194b845b8078eafeae48eb4bf1c9

    |----------------+---------------+-----------------+------------------|
    | DATE           | INTEGER       | DATETIME        | STRUCT           |
    +----------------+---------------+-----------------+------------------+

WILDCARD_QUERY
------------

Description
>>>>>>>>>>>

``wildcard_query(field_expression, query_expression[, option=<option_value>]*)``

> The wildcard_query function maps to the wildcard_query query used in search engine. It returns documents that match provided text in the specified field.

Author: I'm not sure if I should do this. Functions above do not have backticks around their query name in the description.

> The wildcard_query function maps to the wildcard_query query used in search engine. It returns documents that match provided text in the specified field.

Reviewer 2: They should.  For example:

```
QUERY
-----
Description
>>>>>>>>>>>
``query("query_expression" [, option=<option_value>]*)``
The `query` function is an alternative syntax to the `query_string`_ function. It maps to the query_string query used in search engine, to return the documents that match a provided text, number, date or boolean value with a given query expression.
``query_expression`` must be a string provided in Lucene query string syntax. Please refer to examples below:
```

> The wildcard_query function maps to the wildcard_query query used in search engine. It returns documents that match provided text in the specified field.

Author: Thanks. Addressed in 0a4af9dfec1f3a75e9da6a365bd0534ec83c859e

Supported wildcard characters can be found here: https://opensearch.org/docs/latest/opensearch/query-dsl/term/#wildcards

> SQL wildcard % is converted to * as well as _ to ?. You may include % and _ in the search by escaping with a backslash prefix.

Author: Thanks. Addressed in 0a4af9dfec1f3a75e9da6a365bd0534ec83c859e

``wildcard_query(field_expression, query_expression[, option=<option_value>]*)``

> The `wildcard_query` function maps to the `wildcard_query` query used in search engine. It returns documents that match provided text in the specified field.

Reviewer 3: Move entire `wildcard_query` block a bit up - it should be in `RELEVANCE` group

Author: Thanks. Addressed.

OpenSearch supports wildcard characters `*` and `?`.  See the full description here: https://opensearch.org/docs/latest/opensearch/query-dsl/term/#wildcards.

> SQL wildcard character `%` is converted to `*`. SQL character wildcard `_` is converted to `?`. You may include a backslash `\` to escape wildcard characters.

Reviewer 3: You need to escape characters in backticks to ensure that they are interpreted and rendered properly.
https://github.com/Bit-Quill/opensearch-project-sql/blob/1fb497379b5fb7a947e314d0862e59e39af31383/docs/user/dql/functions.rst#wildcard-query

Reviewer 3: I'm not sure that we need to describe in user docs how it works under the hood.
I think you can simplify this line like this:
```
SQL wildcard characters `\%` and `\_` could be escaped by backslash `\\` if you need to perform exact search for those characters.
```
Please, correct wording and validate how escaping in RST works.

Reviewer 3: Consider updating `LIKE` docs about this too.

Author: Thanks. Addressed.

Docs seem to cover like wildcards already https://github.com/opensearch-project/sql/blob/2.x/docs/user/dql/expressions.rst#like
